pygments/lexers/markup.py

```diff
@@ -763,3 +765,720 @@ def _handle_cssblock(self, match):
     def __init__(self, **options):
         self.handlecodeblocks = get_bool_opt(options, 'handlecodeblocks', True)
         RegexLexer.__init__(self, **options)
+
+
+class WikitextLexer(RegexLexer):
+    """
+    For MediaWiki Wikitext.
+
+    Parsing Wikitext is tricky, and results vary between different MediaWiki installations,
+    so we only highlight common syntaxes (built-in or from popular extensions),
+    and also assume templates produce no unbalanced syntaxes.
+    """
+    name = 'Wikitext'
+    url = 'https://www.mediawiki.org/wiki/Wikitext'
+    aliases = ['wikitext', 'mediawiki']
+    filenames = []
+    mimetypes = ['text/x-wiki']
+    flags = re.MULTILINE | re.IGNORECASE
+
+    def nowiki_tag_rules(tag_name):
+        return [
+            (r'(</)({})(\s*)(>)'.format(tag_name), bygroups(Punctuation,
+             Name.Tag, Whitespace, Punctuation), '#pop'),
+            include('entity'),
+            include('text'),
+        ]
+
+    def plaintext_tag_rules(tag_name):
+        return [
+            (r'(?s)(.*?)(</)({})(\s*)(>)'.format(tag_name), bygroups(Text,
+             Punctuation, Name.Tag, Whitespace, Punctuation), '#pop'),
+        ]
+
+    def delegate_tag_rules(tag_name, lexer):
+        return [
+            (r'(</)({})(\s*)(>)'.format(tag_name), bygroups(Punctuation,
+             Name.Tag, Whitespace, Punctuation), '#pop'),
+            (r'(?s).+?(?=</{}\s*>)'.format(tag_name), using(lexer)),
+        ]
+
+    def text_rules(token):
+        return [
+            (r'\w+', token),
+            (r'[^\S\n]+', token),
+            (r'(?s).', token),
```

Comment:
Unfortunately this will cause a lot of regexes to fail. For example, [hr.txt](https://github.com/pygments/pygments/blob/816763b92db1f00f0f200ce3c34fa7c83df6264a/tests/snippets/wikitext/hr.txt) becomes one big text token:
```
'-\n--\n---\n----\n-----\n------\n-------\n--------\n--------' Text
```
instead of being highlighted correctly.